check_api/src/main/java/com/google/errorprone/ErrorProneAnalyzer.java

```diff
@@ -80,13 +80,12 @@ public static ErrorProneAnalyzer createAnalyzer(
                 Suppliers.memoize(
                     () -> {
                       ScannerSupplier toUse =
-                          ErrorPronePlugins.loadPlugins(scannerSupplier, context);
+                          ErrorPronePlugins.loadPlugins(scannerSupplier, context)
+                              .applyOverrides(epOptions);
                       ImmutableSet<String> namedCheckers =
                           epOptions.patchingOptions().namedCheckers();
                       if (!namedCheckers.isEmpty()) {
                         toUse = toUse.filter(bci -> namedCheckers.contains(bci.canonicalName()));
-                      } else {
-                        toUse = toUse.applyOverrides(epOptions);
                       }
```

Comment:
Besides that all existing tests still pass: this change seems reasonable, at least on a conceptual level. I didn't thoroughly assess whether the non-flag changes applied by `applyOverrides` are no-ops, though. Could have a closer look at that later.